allennlp_models/rc/dataset_readers/transformer_squad.py

```diff
@@ -20,41 +20,68 @@
+       `metadata['id']`, `metadata['question']`, `metadata['context']`, `metadata['question_tokens']`,
+       `metadata['context_tokens']`, and `metadata['answers']`. This is so that we can more easily use the
        official SQuAD evaluation script to get metrics.
 
+    For SQuAD v2.0-style datasets that contain impossible questions, we set the gold answer span
+    to the span of the `[CLS]` token when there are no answers.
+
     We also support limiting the maximum length for the question. When the context+question is too long, we run a
-    sliding window over the context and emit multiple instances for a single question. At training time, we only
-    emit instances that contain a gold answer. At test time, we emit all instances. As a result, the per-instance
-    metrics you get during training and evaluation don't correspond 100% to the SQuAD task. To get a final number,
-    you have to run the script in scripts/transformer_qa_eval.py.
+    sliding window over the context and emit multiple instances for a single question.
+    If `skip_invalid_examples` is `True`, then we only emit instances that contain a gold answer.
+    As a result, the per-instance metrics you get during training and evaluation might not correspond
+    100% to the SQuAD task.
+
+    To get a final number for SQuAD v1.1, you have to run
+
+    ```
+    python -m allennlp_models.rc.tools.transformer_qa_eval
+    ```
 
     # Parameters
 
     transformer_model_name : `str`, optional (default=`'bert-base-cased'`)
         This reader chooses tokenizer and token indexer according to this setting.
+
     length_limit : `int`, optional (default=`384`)
         We will make sure that the length of context+question never exceeds this many word pieces.
+
     stride : `int`, optional (default=`128`)
         When context+question are too long for the length limit, we emit multiple instances for one question,
         where the context is shifted. This parameter specifies the overlap between the shifted context window. It
         is called "stride" instead of "overlap" because that's what it's called in the original huggingface
         implementation.
+
     skip_invalid_examples: `bool`, optional (default=`False`)
-        If this is true, we will skip examples that don't have a gold answer. You should set this to True during
-        training, and False any other time.
+        If this is true, we will skip examples where the question+context is truncated according to `length_limit`
+        such that the context no longer contains a gold answer.
+
+        For SQuAD v1.1-style datasets, you should set this to `True` during training, and `False` any other time.
+
+        For SQuAD v2.0-style datasets, leaving this as `False` (recommended) will treat invalid examples
+        the same way as impossible examples, i.e. by setting the answer span to the span of the `[CLS]`
+        token.
```

Comment:
Maybe we should call it "skip impossible instances"? I think SQuAD 2.0 uses the word "impossible" to identify instances without answers.